File: pset1/cash.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

int main(void)
{

//asking the user for the amount of change in dollars
    float dollar;
    do
    {
        dollar = get_float("Enter the amount of Dollars $$: ");
    }
    while (dollar < 0);

//convert the amount to cents
    int cents = round(dollar * 100);

    int penny = 0;
    int nickels = 0;
    int dimes = 0;
    int quarter = 0;

//calaculating how many coins will  be returend
    while (cents > 0)
    {

        while (cents >= 25){
            quarter++;
            cents -= 25;
        }

        while (cents >= 10)
        {
            dimes++;
            cents -= 10;
        }

        while (cents >= 5)
        {
            nickels++;
            cents -= 5;
        }

        while (cents >= 1)
        {
            penny++;
            cents -= 1;
        }
    }

    int coins = penny + nickels + dimes + quarter;
    printf("%i", coins);

}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    dollars = get_float("Enter the amount of Dollars $$: ")

    if dollars > 0:
        break

cents = round(dollars * 100)

coins = 0

while cents > 0:
    if cents >= 25:
        coins += 1
        cents -= 25

    elif cents >= 10:
        coins += 1
        cents -= 10

    elif cents >= 5:
        coins += 1
        cents -= 5

    elif cents >= 1:
        coins += 1
        cents -= 1

print(f"{coins}")